### DISCUSSION TITLE
IceCast2 config

### DISCUSSION THREAD

**Discussion by mcguirepr89** [mcguirepr89/BirdNET-Pi#529] [Q&A] [answered]
Hi, @Binauralhead -- I moved your comment to a new discussion since it was off topic for that discussion.

>Where does Birdnet store its audio config file? I normally use darkice to stream to a few icecast servers and would like to add these to the config file, as an install of birdnet overides the default darkice.cfg on /etc/darkice.cfg
I'm hoping to add my own website to a link on the Birdnet homepage, this saves me having to work out how to use two webserves and two mics in the garden!
>
>Any help much appreciated

**Comment by mcguirepr89** [comment]
Hi, @Binauralhead

The icecast2 config file is located in "`/etc/icecast2/icecast.xml`"

The script that streams the mic to the icecast server is the script `livestream.sh`

Does that information meet your needs?

**Reply by Binauralhead** [reply]
Hi mcguirepr89,

My own icecast server file must be changed too?
Changing anything in the default /etc/darkice.cfg does nolthing after Birdnet install, so it must be in the livestream.sh, but i cant check this just right now as ive taken out the SD card, but I can put it back in again when im ready to faff with my network again as PiHole is running on that Pi, ha!
I'll check out the livestream.sh contents and see where that takes me. 

Thanks for the quick response.
Alan

**Reply by Binauralhead** [reply]
That doesnt seem to help me.
I preinstalled darkice and icecast, the darkice.cfg hed two servers that darkice was streaming to.
After i installed birdnet this config is no longer valid, the darkice.cfg is no longer being used.
How do I change birdnet to point to my original darkice.cfg so that i can still stream to the two servers that were configured in there?
The original icecast.xml i see has been copied to a backup, but the new icecast.xml, i cant seem to figure out where its gettin g the config file from for it settings, like darkce.cfg does.

I'd like to add two servers to the birdnet config so that you can either logon to birdnet and listen to the stream (as it does by default) and have a Locus Sonus stream and a Private server stream all streaming at the same time, the darkice.cfg is very simple to add a new server to, but with birdnet i;m just completley confused as to how it works.

Any help?

Alan

**Reply by mcguirepr89** [reply]
BirdNET-Pi already uses IceCast2 as its liveaudio stream. I don't know much about [DarkIce](http://www.darkice.org/) other the cursory searching and main website, but it seems that it is going to compete with `livestream.sh`, as that script is the one that serves the sound card to the IceCast2 server. 

In general, I don't recommend/support installing BirdNET-Pi on top of other servers. If you tell me more about what you're trying to achieve, I _may_ be able to help, but I'll need clarification on several things:

>I preinstalled darkice and icecast, the darkice.cfg hed two servers that darkice was streaming to.

Please share these configurations, if you don't mind -- I'll have a much better idea about what is not working for you if I know more about how it will affect the system.

>How do I change birdnet to point to my original darkice.cfg so that i can still stream to the two servers that were configured in there?

Could you clarify what you mean here when you say "change birdnet to point to my original darkice.cfg" -- specifically what part of the BirdNET-Pi system will be "pointed" -- perhaps I don't understand how you are hoping to use DarkIce? Any more specifics there would be helpful (e.g., audio driver, transcoder used, ports/sockets it might need, protocol used).

>The original icecast.xml i see has been copied to a backup, but the new icecast.xml, i cant seem to figure out where its gettin g the config file from for it settings, like darkce.cfg does.

The best info I can think to give at this stage is to again mention the IceCast2 configuration file "`/etc/icecast2/icecast.xml`", the `livestream.sh` script that streams the audio card input to the IceCast2 server, and your `birdnet.conf` file that has your configuration for your sound card (`REC_CARD`) and channels (`CHANNELS`), and possibly the `/etc/caddy/Caddyfile` Caddy configuration file that acts as the reverse proxy to access the server. That is the entire stack that serves the live audio stream. **Note:**The `ICE_PWD` cannot be set in `birdnet.conf` post-install -- it is only used/set during installation. Any changes there won't change the actual IceCast2 server settings.

>I'd like to add two servers to the birdnet config so that you can **either logon to birdnet and listen to the stream (as it does by default)** and [have a Locus Sonus stream and a Private server stream all streaming at the same time . . .]()

I'm afraid I don't understand. I do see the "**either**" clause, but am confused about the "[or]()"

**Reply by Binauralhead** [reply]
Hi,

Darkice config as follows, configured for two servers that was installed before birdnet, Icecast2 was installed prior to birdnet to, so that I can locally serve to the LAN if I so wished. I'm still a little confused as to how icecast works because I simply use darkice to forward any streams and only use icecast for serving the local LAN if I want.

[icecast2-0]
bitrateMode = cbr 
bitrate =  320 
format = mp3
quality = 1.0
server = solid2.streamupsolutions.com  
port =  11124
mountPoint = live
password = *********
name = sr3d-live
description = Live binaural stream for headphones
genre = Binaural 3D for headphones
public = yes

[icecast2-1]
bitrateMode = cbr
bitrate =  320
quality = 1.0
format = mp3
server = locus.creacast.com
port =  9001
password = ********
mountPoint = liverpool_skelmersdale.mp3
name = Liverpool - Skelmersdale
description = SR3D Live binaural dummy head 3D audio stream for headphones
url = https://www.sr3d.co.uk/store/p10/SR3D_Dummy_head.html
genre = Binaural 3D audio for headphones
public = yes

#[icecast2-2]
#bitrateMode = cbr
#bitrate =  320
#format = mp3
#quality = 1.0
#server = localhost
#port = 8000
#mountPoint = live
#password = hackme
#name = sr3dlive
#description = Live binaural stream for headphones
#genre = Binaural 3D for headphones
#public = yes


I looked at the index.php on birdnet and see that I cant use my own website and have it as a link, or have a link on birdnet to my site, it just shows a unobtainable page for birdnet, so thats out the window I guess, I was hoping to utilise one raspberry pi with my binaural recordings I put on there now and then with a link to the birdnet website on the pi.

There's a "play live audio" button on the birdnet page, so where ever that server info is, I was hoping to also add the two other servers so that they could all be streaming at the same time, just like darkice allows, I guess its all in the icecast.xml file, but Im not accustomed to that file, or configurations.

Reverse proxies are over my head - re: caddy file so realy green there am afraid.

Sooo, darkice is a seperate server, and icecast2 is a seperate server? I didnt know icecast could send an audio stream from a sound card, I thought you needed darkice to do that for you. 
Is it possible to use darkice to send to birdnet, so I can simply just add it to the darkce.cfg? this looks like its going to be a bit harder than I thought.. doh!

Hope I didnt miss anything

Alan

**Comment by mcguirepr89** [comment, marked as answer]
This isn't supported.
